Nucleation_model.py

```diff
@@ -87,7 +87,8 @@ def residual(t, solution):
 
 nucleations = solution.y[0]
 area = solution.y[1]
-Li_concentration = solution.y[2]
+radius = solution.y[2]
+Li_concentration = solution.y[3]
```

Comment:
So these types of changes would be handled by defining a dictionary of pointers somewhere in the model initialization.

So instead of calling `y[2]` or whatever, you would call `y[ptr['Li']]` or somesuch, where `ptr['Li']` is set in one place in your code.  Now if the order changes, you just go and change that place where `ptr['Li']` is set....